test/extensions/transport_sockets/proxy_protocol/proxy_protocol_integration_test.cc

```diff
@@ -681,5 +705,55 @@ TEST_P(ProxyProtocolTLVsIntegrationTest, TestV2ProxyProtocolPassWithTypeLocal) {
   ASSERT_TRUE(fake_upstream_connection_->waitForDisconnect());
 }
 
+TEST_P(ProxyProtocolTLVsIntegrationTest, TestV2TLVProxyProtocolWithCustomMetadata) {
```

Comment:
We might also want an integration test for 1.) passthrough + custom TLVs, or 2.) no downstream PP2 listener/network filter, but upstream transport socket is PP2.

Reply:
Added an integration test for passthrough + custom TLVs in https://github.com/timflannagan/envoy/pull/1/commits/974bc574b4334048e39d408716e5aa4352205e83.